### PR TITLE
Feature/be able to use dash

### LINKED ISSUE #188: Resource name cannot contain dashes
## Is your feature request related to a problem?

Resources with a dash are not taken into account. So it generates error messages when using this plugin.

```
2019/11/24 09:56:35 [ERROR] There was an error initialising the terraform provider: plugin terraform-provider-netbox init error while creating schema provider: resource name can not be empty
```

## Describe the solution you'd like

I would like that resource name with dashes are taken into account.

## Acceptance criteria

No more error message displayed when using swagger.json with resource names containing one or more dashes. These kind of resources are recognized by the plugin.

## Describe alternatives you've considered

None.

## Additional context

None.

## Checklist

**Don't forget to go through the checklist to make sure the issue is created properly**:

- [X] I have added a corresponding label (feature request/bug/etc) to the issue (right side menu)
- [X] I have added this issue to the 'API Terraform Provider' GitHub project (right side menu)

### PR DESCRIPTION
**Pull request by smutel**
## Proposed changes

Currently dashes in resource name are not accepted.

## Type of change

- [ ] Bug-fix (change that fixes current functionality)
- [x] New feature (change that adds new functionality)

=> Resolve #188

## Checklist

Please put an `x` (w/o heading/trailing white spaces) in the boxes that apply:

- [X] I have read and followed the [CONTRIBUTING](https://github.com/dikhan/terraform-provider-api/blob/master/.github/CONTRIBUTING.md) guidelines
- [X] I have made sure code compiles correctly
- [X] I have run 'make test' locally from the terraform_provider_api folder and no errors were found
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added/updated necessary documentation (if appropriate)

### CODE REVIEW

**Comment by dikhan** [comment]
Hi @smutel ,

Thanks for creating the PR! Would you mind opening an issue that describes the problem this PR solves? I see the PR contains a brief summary, so perhaps you can reuse that and also provide some more info in terms on what the acceptance criteria is etc. 

To provide more context on why having the issue described is important, it helps keeping track of the problems we are trying to solve, prioritise them accordingly, have some history that we can refer back to and also the new versions of the plugin released have links to the issues fixed too. 

I will try to take a look at the PR sometime this week.

File: openapi/openapi_v2_resource_test.go [hunk 1]
```diff
@@ -484,6 +484,12 @@ func TestBuildResourceNameFromPath(t *testing.T) {
 			expectedResourceName: "cdns",
 			expectedError:        nil,
 		},
+		{
+			path:                 "/cdns-test",
+			preferredName:        "",
+			expectedResourceName: "cdns-test",
```
Comment (dikhan) [inline openapi/openapi_v2_resource_test.go]: For consistency reasons and to abide by Terraform snake_case naming conventions, can the expectedResourceName be updated to cdns_test?

Comment (smutel) [reply openapi/openapi_v2_resource_test.go]: But the dash will be replaced by an underscore by the plugin ?
Currently the tests are passing so I am a little bit confused ...

Comment (dikhan) [reply openapi/openapi_v2_resource_test.go]: yeah that'd be a good option, perhaps replacing the dashes with underscores and updating the tests to reflect that.

Comment (smutel) [reply openapi/openapi_v2_resource_test.go]: Done

**Comment by dikhan** [comment]
Thanks for contributing @smutel !

**Pull request merged**